packages/d3fc-webgl/src/scale/scaleMapper.js

```diff
@@ -0,0 +1,32 @@
+import d3Scale from 'd3-scale';
+import glScaleLinear from './glScaleLinear';
+import glScaleLog from './glScaleLog';
+import glScalePow from './glScalePow';
+
+export default (scale) => {
+    let outScale = d3Scale.scaleIdentity();
+    let glScale = null;
+    const scaleCopyString = scale.copy.toString();
+
+    if (scaleCopyString === d3Scale.scaleLinear().copy.toString()) {
```

Comment:
Could you add a comment here when you get chance, I'm not sure I follow the logic of making all these string copies

Reply:
Sure, effectively it's a bit of a hacky way of telling which type of d3 scale is passed in, since there's no `scale.name` / `scale.type` to compare against. `scale.copy` gives a function which is specific to the scale type - it's horrible but it works for now.

I'm very much eager to replace this with something nicer.